persistence/module.go

```diff
@@ -106,9 +106,14 @@ func (*persistenceModule) Create(bus modules.Bus, options ...modules.ModuleOptio
 	treeModule, err := trees.Create(
 		bus,
 		trees.WithTreeStoreDirectory(persistenceCfg.TreesStoreDir),
-		trees.WithLogger(m.logger))
+		trees.WithLogger(m.logger),
+		trees.WithTxIndexer(txIndexer))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to create TreeStoreModule: %w", err)
+	}
+	treeStoreModule, ok := treeModule.(modules.TreeStoreModule)
```

Comment:
Can you explain why this is needed? does not `trees.Create` return `modules.TreeStoreModule`?